server/.go-version

```diff
@@ -1 +1 @@
-1.20.7
\ No newline at end of file
+1.21.8
```

Comment:
Unfortunately we need to keep this until we stop supporting Mattermost versions with go versions before 1.21 . 
Our packaging pipeline depends on that file and it will mess the functionality for the previous supported releases . 
I will add a reminder for `2024-11-15` when we will stop supporting 9.5 release 🚀 
cc: @mattermost/core-build-engineers